Unit test failure on OSX but not on linux

```
$ mvn install
...
[INFO] Running sparkling.GenericIntegrationTest
[Thread-1] INFO org.eclipse.jetty.util.log - Logging initialized @222ms to org.eclipse.jetty.util.log.Slf4jLog
[Thread-1] WARN org.eclipse.jetty.server.AbstractConnector - Ignoring deprecated socket close linger time
== Sparkling has ignited ...
>> Listening on 0.0.0.0:4567
[Thread-1] INFO org.eclipse.jetty.server.Server - jetty-9.4.55.v20240627; built: 2024-06-27T15:05:37.765Z; git: 70015831e5195ba74da6383c4ab7c90c56bb08af; jvm 11.0.25+0
[Thread-1] INFO org.eclipse.jetty.server.session - DefaultSessionIdManager workerName=node0
[Thread-1] INFO org.eclipse.jetty.server.session - No SessionScavenger set, using defaults
[Thread-1] INFO org.eclipse.jetty.server.session - node0 Scavenging every 660000ms
[Thread-1] INFO org.eclipse.jetty.server.AbstractConnector - Started ServerConnector@5e5a570e{HTTP/1.1, (http/1.1)}{0.0.0.0:4567}
[Thread-1] INFO org.eclipse.jetty.server.Server - Started @273ms
Body was: Fo shizzy
Body was: Fo shizzy
external directory: /var/folders/15/qrc0v_5d3v90r31w29kb3q_h0000gn/T/sparklingGenericIntegrationTest17880698820882847702
external file: /var/folders/15/qrc0v_5d3v90r31w29kb3q_h0000gn/T/sparklingGenericIntegrationTest17880698820882847702/externalFile.html
>>> Sparkling shutting down...[main] INFO org.eclipse.jetty.server.AbstractConnector - Stopped ServerConnector@5e5a570e{HTTP/1.1, (http/1.1)}{0.0.0.0:4567}
[main] INFO org.eclipse.jetty.server.session - node0 Stopped scavenging
done
[ERROR] Tests run: 16, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.765 s <<< FAILURE! -- in sparkling.GenericIntegrationTest
[ERROR] sparkling.GenericIntegrationTest.testExternalStaticFile -- Time elapsed: 0.004 s <<< FAILURE!
junit.framework.AssertionFailedError: expected:<200> but was:<404>
```

--- a/sparkling-template/README.md
+++ b/sparkling-template/README.md
@@ -14,7 +14,7 @@ Template routes for Sparkling web framework:
 ```
 
 
-#### Hacking sparking-template
+### Hacking sparking-template
 
 To build
 
@@ -44,7 +44,8 @@ http://localhost:4567/hello/foo
 in a browser.
 
 
-#### Extending sparkling-template
+
+### Extending sparkling-template
 
 To add a new Template engine
 

--- a/sparkling/src/test/java/sparkling/GenericIntegrationTest.java
+++ b/sparkling/src/test/java/sparkling/GenericIntegrationTest.java
@@ -11,11 +11,14 @@ import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
 
+import java.security.SecureRandom;
+
 import static sparkling.Sparkling.*;
 
 public class GenericIntegrationTest {
 
     static SparklingTestUtil testUtil;
+    static File tmpExternalDirectory;
     static File tmpExternalFile;
 
     @AfterClass
@@ -25,13 +28,31 @@ public class GenericIntegrationTest {
         if (tmpExternalFile != null) {
             tmpExternalFile.delete();
         }
+        if (tmpExternalDirectory != null) {
+            tmpExternalDirectory.delete();
+        }
+    }
+
+    // see TempFileHelper.java
+    private static final SecureRandom random = new SecureRandom();
+    private static String generateTempDirectoryName(final String prefix) {
+        long n = random.nextLong();
+        return prefix + Long.toUnsignedString(n);
     }
 
     @BeforeClass
     public static void setup() throws IOException {
         testUtil = new SparklingTestUtil(4567);
 
-        tmpExternalFile = new File(System.getProperty("java.io.tmpdir"), "externalFile.html");
+        //
+        // does not work on osx
+        // see https://github.com/jetty/jetty.project/security/advisories/GHSA-g3wg-6mcf-8jj6
+        //
+        //tmpExternalDirectory = Files.createTempDirectory("sparklingGenericIntegrationTest").toFile();
+        tmpExternalDirectory = new File(generateTempDirectoryName("work"));
+        tmpExternalDirectory.mkdir();
+
+        tmpExternalFile = new File(tmpExternalDirectory, "externalFile.html");
 
         FileWriter writer = new FileWriter(tmpExternalFile);
         writer.write("Content of external file");
@@ -39,7 +60,7 @@ public class GenericIntegrationTest {
         writer.close();
 
         staticFileLocation("/public");
-        externalStaticFileLocation(System.getProperty("java.io.tmpdir"));
+        externalStaticFileLocation(tmpExternalDirectory.getAbsolutePath());
 
         before(new Filter("/protected/*") {
 
@@ -210,13 +231,13 @@ public class GenericIntegrationTest {
     @Test
     public void testTwoRoutesWithDifferentCaseButSameName() {
         String lowerCasedRoutePart = "param";
-        String uppperCasedRoutePart = "PARAM";
+        String upperCasedRoutePart = "PARAM";
 
         registerEchoRoute(lowerCasedRoutePart);
-        registerEchoRoute(uppperCasedRoutePart);
+        registerEchoRoute(upperCasedRoutePart);
         try {
             assertEchoRoute(lowerCasedRoutePart);
-            assertEchoRoute(uppperCasedRoutePart);
+            assertEchoRoute(upperCasedRoutePart);
         } catch (Throwable e) {
             throw new RuntimeException(e);
         }

